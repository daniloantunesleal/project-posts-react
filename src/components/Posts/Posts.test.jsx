import { Posts } from "."
import { render, screen } from "@testing-library/react"

const props = {
    posts: [
        {
            title: "this is the title",
            body: "this is the body",
            id: 1,
            cover: "img/daniloaleal.png"
        },
        {
            title: "this is the title 2",
            body: "this is the body 2",
            id: 2,
            cover: "img/daniloaleal2.png"
        },
        {
            title: "this is the title 3",
            body: "this is the body 3",
            id: 3,
            cover: "img/daniloaleal3.png"
        }
    ]
}

describe("<Posts />", () => {
    it("should render posts", () => {
        render(<Posts {...props} />)

        expect(screen.getAllByRole("heading", { name: /this is the title/i })).toHaveLength(props.posts.length)
        expect(screen.getAllByRole("img", { name: /this is the title/i })).toHaveLength(props.posts.length)
        expect(screen.getAllByText(/this is the body/i)).toHaveLength(props.posts.length)
    
    })

    it("should not render posts", () => {
        render(<Posts />)

        expect(screen.queryAllByRole("heading", { name: /this is the title/i })).not.toBeInTheDocument()
    })

    it("should match snapshot", () => {
        const { container } = render(<Posts {...props} />)

        // eslint-disable-next-line testing-library/no-node-access
        expect(container.firstChild).toMatchSnapshot()
    
    })
})