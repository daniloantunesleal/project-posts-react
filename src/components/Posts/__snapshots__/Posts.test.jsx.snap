// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posts /> should match snapshot 1`] = `
<div
  class="posts"
>
  <div
    class="post"
  >
    <img
      alt="this is the title"
      src="img/daniloaleal.png"
    />
    <div
      class="post-content"
    >
      <h2>
        this is the title
      </h2>
      <p>
        this is the body
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="this is the title 2"
      src="img/daniloaleal2.png"
    />
    <div
      class="post-content"
    >
      <h2>
        this is the title 2
      </h2>
      <p>
        this is the body 2
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="this is the title 3"
      src="img/daniloaleal3.png"
    />
    <div
      class="post-content"
    >
      <h2>
        this is the title 3
      </h2>
      <p>
        this is the body 3
      </p>
    </div>
  </div>
</div>
`;
